diff --git a/proto/tt_top.v b/proto/tt_top.v
index 94385a7..535c149 100644
--- a/proto/tt_top.v
+++ b/proto/tt_top.v
@@ -21,6 +21,9 @@ module tt_top #(
 	output wire [N_PADS-1:0] io_out,
 	output wire [N_PADS-1:0] io_oeb,
 
+	// Debug: selected design
+	output wire [9:0] si_sel,
+
 	// From caravel
 	input  wire user_clock2,
 
@@ -102,6 +105,8 @@ module tt_top #(
 		{ UNUSED_PADS{k_one} }
 	};
 
+	assign si_sel = spine_iw[11:2];
+
 	// Tie-offs
 		// Control High
 	assign pad_ch_out  = { k_zero, k_zero, k_zero, k_zero, k_zero, k_zero };
